pkg/machinery/resources/hardware/system_information.go

```diff
@@ -0,0 +1,85 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package hardware
+
+import (
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+	"github.com/cosi-project/runtime/pkg/resource/typed"
+)
+
+// SystemInformationType is type of SystemInformation resource.
+const SystemInformationType = resource.Type("SystemInformation.hardware.talos.dev")
+
+// SystemInformation resource holds node SystemInformation information.
+type SystemInformation = typed.Resource[SystemInformationSpec, SystemInformationRD]
+
+// SystemInformationSpec represents the system information obtained from smbios.
+type SystemInformationSpec struct {
```

Comment:
we should register new resource in the `v1alpha2_state.go` to make it work with `talosctl get`